README.md

```diff
@@ -409,56 +409,36 @@ Components](https://github.com/styled-components/styled-components) and
 Purview components do not run in the browser.
 
 ## Error handling
-Uncaught errors within Purview will bubble up as
-[`uncaughtException`][uncaught-exception] or
-[`unhandledRejection`][unhandled-rejection] events. These can be caught and
-handled by registering event listeners on `process`.
+Normally, when an error occurs in the context of a Purview component, it will 
+appear as top-level [`uncaughtException`][uncaught-exception] or
+[`unhandledRejection`][unhandled-rejection] events. Purview provides the option
+to observe or modify these errors with a custom error handler before they
+appear within these top-level events.
 
-Additionally, certain classes of errors can be instrumented in Purview prior to
-going to these top-level events:
+In order to add a custom error handler, pass an `onError` function as an option
+to `Purview.render()`. When provided, the handler is invoked for these classes
+of Purview errors:
 
 - Errors within event callbacks.
 - Errors in the component render path.
 - Errors when setting initial component state, after WebSocket connection is
 established.
 
-Register an instrumentation function by passing it in as an option parameter to
-`Purview.render()`:
+If your handler does not itself throw an error, Purview will re-throw the error
+in order to prevent execution of further code.
 
 ```tsx
-const app = express()
-
 app.get("/", async (req, res) => {
   function onError(error: unknown): void {
```

Comment:
can we type the `error` as `Error`?

Reply:
This is done, but this reminded me of why I typed it as `unknown` in the first place.

In TypeScript v4+ where the type of error within `catch` blocks is [changed from `any` to `unknown`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#unknown-on-catch-clause-bindings) in strict mode.

I don't think it matters for our use case but I think it may actually be safer to change type of `ErrorHandler` from `(error: Error) => void` to `(error: unknown) => void` instead:
https://github.com/shirakomamu/purview/blob/2dbed627f50fd417141fefdd71beb4dbb65d363a/src/purview.ts#L120

I don't have a strong opinion either way though, so if `Error` seems fine with you then as-is should be good to go.

Reply:
@karthikv Sure, that sounds good! I've changed the type back to `unknown` in latest commit.

I uncovered an issue with this typing when the user throws non-objects. Since WeakSets can only hold objects, there was a runtime error when the error is actually a string, such as `throw "Error Message"` (as opposed to something like `throw new Error("Error Message")`.

I added a check to where we assign the error handler to the root here to only add the error to the WeakSet if it's an object to account for this case:
https://github.com/karthikv/purview/blob/7b14a1c5b90328e773245262001652a289df67f7/src/purview.ts#L570-L580

With this change, non-object errors will call the handler multiple times in nested paths, but I think that can be something the consumer should be aware of and handle themselves when writing the handler.